com.ibm.wala.dalvik/src/com/ibm/wala/dalvik/classLoader/DexFileModule.java

```diff
@@ -73,6 +73,7 @@
 	private final File f;
     private final DexFile dexfile;
     private final Collection<ModuleEntry> entries;
+    public final static int API_LEVEL = 28;  // Android API level
```

Comment:
Yes, it could cause issues. But also API level 24 might have the same issue. As far as I understand the dexlib code, the best option is to not provide any specific api level at all and let dexlib figure it out. 

However, this is not possible as DexFileFactory.loadDexContainer requires the opcodes to be nonNull while other methods like loadDexFile/loadDexEntry allow it to be null. I think for loadDexContainer it should be nullable as well as it internally uses it for the exact same code. So probably this is a (minor) bug in the dexlib code.

Reply:
Can we report a bug to dexlib?  And also, could we allow for passing in a parameter to override the API level in our code?

Reply:
I posted a PR: https://github.com/JesusFreke/smali/pull/673
I agree that having both options is a good idea. Let's wait for a reply.

Reply:
The PR has just been merged and there is a new tagged release 2.2.6

Reply:
Looks like many previous releases are on Maven central.  Maybe we can ping the dexlib2 author and ask about it?